src/libs/actions/templates.ts

```diff
@@ -33,7 +33,10 @@ export async function UpsertTemplate({
           templateStyle,
           templateSampleData: templateSampleData ? JSON.parse(templateSampleData) : {},
         })
-        .where(eq(templates.id, templateId));
+        .where(and(
+          eq(templates.templateId, templateId),
+          eq(templates.environment, 'dev'),
+        ));
 
       return { success: true, message: 'Template updated successfully' };
     } else {
```

Comment:
no need of else

Reply:
use this https://orm.drizzle.team/docs/guides/upsert

instead of this condition